conda_build/conda_interface.py

```diff
@@ -135,3 +136,160 @@ def which_prefix(path):
+                return cls[value]
+            return super(PathType, cls).__call__(value, *args, **kwargs)
+
+        @classmethod
+        def __getitem__(cls, name):
+            return cls._member_map_[name.replace('-', '').replace('_', '').lower()]
+
+        def __int__(self):
+            return self.value
+
+        def __str__(self):
+            return self.name
+
+        def __json__(self):
+            return self.name
+
+
+    class FileMode(Enum):
+        """
+        Refers to the mode of the file. Originally referring to the has_prefix file, but adopted
+        for paths.json
+        """
+        text = 'text'
+        binary = 'binary'
+
+        def __str__(self):
+            return "%s" % self.value
+
+
+    class EntityEncoder(JSONEncoder):
+        # json.dumps(obj, cls=SetEncoder)
+        def default(self, obj):
+            if hasattr(obj, 'dump'):
+                return obj.dump()
+            elif hasattr(obj, '__json__'):
+                return obj.__json__()
+            elif hasattr(obj, 'to_json'):
+                return obj.to_json()
+            elif hasattr(obj, 'as_json'):
+                return obj.as_json()
+            return JSONEncoder.default(self, obj)
+
+
+    # work-around for python bug on Windows prior to python 3.2
+    # https://bugs.python.org/issue10027
+    # Adapted from the ntfsutils package, Copyright (c) 2012, the Mozilla Foundation
+    class CrossPlatformStLink(object):
+        _st_nlink = None
+
+        def __call__(self, path):
+            return self.st_nlink(path)
+
+        @classmethod
+        def st_nlink(cls, path):
+            if cls._st_nlink is None:
+                cls._initialize()
+            return cls._st_nlink(path)
+
+        @classmethod
+        def _standard_st_nlink(cls, path):
+            return lstat(path).st_nlink
+
+        @classmethod
+        def _windows_st_nlink(cls, path):
+            st_nlink = cls._standard_st_nlink(path)
+            if st_nlink != 0:
+                return st_nlink
+            else:
+                # cannot trust python on Windows when st_nlink == 0
+                # get value using windows libraries to be sure of its true value
+                # Adapted from the ntfsutils package, Copyright (c) 2012, the Mozilla Foundation
+                GENERIC_READ = 0x80000000
+                FILE_SHARE_READ = 0x00000001
+                OPEN_EXISTING = 3
+                hfile = cls.CreateFile(path, GENERIC_READ, FILE_SHARE_READ, None,
+                                       OPEN_EXISTING, 0, None)
+                if hfile is None:
+                    from ctypes import WinError
+                    raise WinError()
+                info = cls.BY_HANDLE_FILE_INFORMATION()
+                rv = cls.GetFileInformationByHandle(hfile, info)
+                cls.CloseHandle(hfile)
+                if rv == 0:
+                    from ctypes import WinError
+                    raise WinError()
```

Comment:
Please don't raise this error with no message.  Try to explain when went wrong here - that creating the file failed at least, and any other info that might be helpful